src/client/pool.go

```diff
@@ -66,6 +72,12 @@ func (p *Pool) Request(request *Request) *Request {
 
 // Send adds request to pool.
 func (p *Pool) Send(request *Request) {
+	// Check if a free space is in the "requestsChan" channel.
+	// If not, then a deadlock would occur. See test "TestPoolManyRequests".
+	if !p.started && p.requestsQueuedCount.IncAndGet() >= REQUESTS_BUFFER_SIZE {
+		panic(fmt.Errorf(`Too many (%d) queued reuests in HTTP pool.`, p.requestsQueuedCount.Get()))
```

Comment:
Okrem toho sa to tu kontroluje, ... ak `pool` este neodstartoval, tak nie je mozne vlozit viac ako je maximum, ... tj. kod sa tu nikdy nezablokuje.

Ak je `pool` spusteny, tak sa kod moze zablokovat, kedze casom nejaky request dobehne a uvolni sa miesto.